neo/Consensus/ConsensusService.cs

```diff
@@ -385,6 +496,10 @@ private void OnTransaction(Transaction transaction)
             if (transaction.Type == TransactionType.MinerTransaction) return;
             if (!context.State.HasFlag(ConsensusState.Backup) || !context.State.HasFlag(ConsensusState.RequestReceived) || context.State.HasFlag(ConsensusState.ResponseSent) || context.State.HasFlag(ConsensusState.ViewChanging) || context.State.HasFlag(ConsensusState.BlockSent))
                 return;
+            // If we are changing view but we already have enough preparation payloads to commit in the current view,
+            // we must keep on accepting transactions in the current view to be able to create the block.
+            if (context.State.HasFlag(ConsensusState.ViewChanging) &&
+                context.PreparationPayloads.Count(p => p != null) < context.M) return;
```

Comment:
Maybe you should remove L497's `context.State.HasFlag(ConsensusState.ViewChanging)` first?

Reply:
Oops. I had removed it I thought. Maybe I missmerged it from my other local repository I've been going back and forth with.

Reply:
Fixed.

Reply:
I like this logic, it solves the case for that node to still enter in Commit phase.
However, It helps when we already have `M` Preparations.
But if we have `M - 1`. What would be the drawback of keep accepting transactions and try to help the achievement of Commitment meanwhile `M` ChangeViews were not accomplished?